bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/BKTConfig.kt

```diff
@@ -0,0 +1,111 @@
+package io.bucketeer.sdk.android
+
+import io.bucketeer.sdk.android.internal.logw
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+
+internal const val DEFAULT_FLUSH_INTERVAL_MILLIS: Long = 30_000 // 30 seconds
+internal const val DEFAULT_MAX_QUEUE_SIZE: Int = 50
+internal const val DEFAULT_POLLING_INTERVAL_MILLIS: Long = 600_000 // 10 minutes
+internal const val DEFAULT_BACKGROUND_POLLING_INTERVAL_MILLIS: Long = 3_600_000 // 1 hour
+internal const val MIN_POLLING_INTERVAL_MILLIS: Long = 300_000 // 5 minutes
+internal const val MIN_BACKGROUND_POLLING_INTERVAL_MILLIS: Long = 1_200_000 // 20 minutes
+
+data class BKTConfig internal constructor(
+  val apiKey: String,
+  val endpoint: String,
+  val featureTag: String,
+  val eventsFlushInterval: Long,
+  val eventsMaxBatchQueueCount: Int,
+  val pollingInterval: Long,
+  val backgroundPollingInterval: Long,
+  val debugMode: Boolean
```

Comment:
I've been thinking that maybe we don't need this flag.
I thought that we could use it around here:

https://github.com/ca-dp/bucketeer-android-sdk/blob/master/bucketeer/src/main/java/jp/bucketeer/sdk/Bucketeer.kt#L88-L92

But creating a new option for the config only for that is a bit too much.
Do you think there will be other uses for this flag?

Reply:
No, that's the only use I can think of for now.

But then how do you change logger?
Maybe add `fun looger(logger: Logger)` instead of `debugMode(debugMode: Boolean)`?

Reply:
Yes, I think it will be better!
Let's go with the `logger ` function, then.

Reply:
I've added `BKTLogger` interface in [`a7e08bd` (#61)](https://github.com/ca-dp/bucketeer-android-sdk/pull/61/commits/a7e08bd06ac9008c1b4ec431cdb340926ea64f76)